app/main/routes.py

```diff
@@ -9,9 +14,51 @@ def index():
     return redirect(url_for("main.upload"))
 
 
-@bp.route("/upload", methods=["GET"])
+@bp.route("/upload", methods=["GET", "POST"])
 def upload():
-    return render_template("upload.html")
+
+    with open("govuk_components/file-upload/fixtures.json") as json_file:
+        fixtures = json.load(json_file)
+
+    if request.method == "GET":
+        return render_template("upload.html", fixtures=fixtures['fixtures'], response='')
+
+    if request.method == "POST":
+        excel_file = request.files.get('file-upload-1')
+        response = post_ingest(current_app.config.get('DATA_STORE_API_HOST'),
+                      '/ingest',
+                      excel_file)
+
+        return render_template("upload.html", fixtures=fixtures['fixtures'], response=response.json())
+
+
+def post_ingest(hostname: str, endpoint: str, excel_file: object) -> Response:
+    """Send an HTTP POST request to ingest into the data store server and return the response.
+
+    This function constructs the request URL using the provided hostname, endpoint.
+    It sends a HTTP POST request including the given excel file to the constructed URL and handles any bad responses.
+
+    :param hostname: The hostname or base URL of the remote server.
+    :param endpoint: The endpoint or path of the remote resource to retrieve.
+    :param excel_file: Excel workbook uploaded to the front end
+    :return: The requests Response object containing the response from the remote server.
+    """
+    request_url = hostname + endpoint
+    excel_mimetype = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+    files = {"excel_file": (excel_file.name, excel_file, excel_mimetype)}
+
+    response = requests.post(request_url,
+                             files=files,
+                             data={'source_type': 'tf_round_three'})
+
+    if response.status_code in [200, 400]:
+        return response
+
+    else:
+        current_app.logger.error(
+            f"Bad response: {request_url} returned {response.status_code}"
+        )
+        return abort(500)
```

Comment:
Put this function in a separate file. In routes.py, we only want functions explicitly involved in routing.